tests/components/delete_channel_modal.test.jsx

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) 2017-present Mattermost, Inc. All Rights Reserved.
+// See License.txt for license information.
+
+import React from 'react';
+import {shallow} from 'enzyme';
+
+import DeleteChannelModal from 'components/delete_channel_modal/delete_channel_modal.jsx';
+
+describe('components/delete_channel_modal', () => {
+    test('should match snapshot for delete_channel_modal', () => {
+        function emptyFunction() {} //eslint-disable-line no-empty-function
+        const wrapper = shallow(
+            <DeleteChannelModal
+                onHide={emptyFunction}
+                currentTeamDetails={{
+                    name: 'mattermostDev'
+                }}
+                channel={{
+                    create_at: 1508265709607,
+                    creator_id: 'zaktnt8bpbgu8mb6ez9k64r7sa',
+                    delete_at: 0,
+                    display_name: 'testing',
+                    extra_update_at: 1508265709628,
+                    header: 'test',
+                    id: 'owsyt8n43jfxjpzh9np93mx1wa',
+                    last_post_at: 1508265709635,
+                    name: 'testing',
+                    purpose: 'test',
+                    team_id: 'eatxocwc3bg9ffo9xyybnj4omr',
+                    total_msg_count: 0,
+                    type: 'O',
+                    update_at: 1508265709607
+                }}
+                actions={{
+                    deleteChannel: emptyFunction
+                }}
+            />
+        );
+        expect(wrapper).toMatchSnapshot();
+    });
+});
```

Comment:
All looks good to me except for the test coverage which is low (see screenshots). I'd like to request that the functions: `handleDelete()`, `onHide()`, and `handleKeyDown(e)` are well covered. Let me know if you could push another commit, otherwise I'll push separate patch to have those covered. 

If you're looking for a pattern, here are some good examples like:
- https://github.com/mattermost/mattermost-webapp/blob/master/tests/components/integrations/installed_command.test.jsx, or
- https://github.com/mattermost/mattermost-webapp/blob/master/tests/components/edit_channel_purpose_modal.test.jsx

![screen shot 2017-10-20 at 12 29 10 am](https://user-images.githubusercontent.com/5334504/31782987-4c070a84-b52f-11e7-850c-cbb71222d4e1.png)

![screen shot 2017-10-20 at 12 29 24 am](https://user-images.githubusercontent.com/5334504/31783005-56e9b898-b52f-11e7-968d-67cec7e2f120.png)

Reply:
@saturninoabril tried that. I had an issue with browserHistory.push as it was not on browser. I could't do it I tried `withHistory` LOC and with `createMemoryHistory` so gave up. Any suggestions?

Reply:
@saturninoabril Yes done. Thanks
![screen shot 2017-10-20 at 9 21 54 am](https://user-images.githubusercontent.com/4973621/31804643-3b3e8a58-b578-11e7-8456-2cd77e5826a3.png)

Reply:
Great, thanks!